pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -97,3 +103,21 @@ func keyPairPEM() (string, string, error) {
 
 	return pubKeyPEM.String(), privKeyPEM.String(), nil
 }
+
+func localJWTForKey(id string, privateKkeyPem string) (string, error) {
+	priv, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKkeyPem))
+	if err != nil {
+		return "", err
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, jwt.MapClaims{
+		string(InfraEnvKey): id,
```

Comment:
To help with code maintenance and for developers who aren't familiar with JWT, it would be nice to explain why the infraenv id is used to generate the JWT token versus creating a token with no claims.

Reply:
Sure, Richard. I will add some code comments to explain this in the upcoming PRs. Meanwhile, here is the explanaiton.

The code embeds a value, an infra-env id, which is unique for every cluster into the JWT token using a specific key name (InfraEnvKey). When the token is received and parsed at the assisted-service side, the code extracts this ID using the same key (InfraEnvKey). This allows the application to verify the token's origin and potentially tailor its behavior or access control based on the specified environment.

Overall, InfraEnvKey provides a mechanism to link JWT tokens with specific infrastructure environments, enhancing security. The implementation of a new authenticator in assisted service via upcoming PR is based on the existing local authenticator. Here is a ref https://github.com/openshift/assisted-service/blob/master/pkg/auth/local_authenticator.go#L67-L80